123_state_tasks.R

```diff
@@ -17,11 +17,30 @@ do_state_tasks <- function(oldest_active_sites, ...) {
      sprintf("get_site_data(sites_info_file = '1_fetch/tmp/inventory_%s.tsv', parameter = parameter)", task_name)
    }
 )
-
+  # New step to create plots
+  plot_step <- create_task_step(
+    step_name = 'plot',
+    target_name = function(task_name, step_name,  ...){
+      sprintf("3_visualize/out/timeseries_%s.png", task_name)
+    },
+    command = function(steps, ...){
+      sprintf("plot_site_data(out_file = target_name, site_data = %s, parameter=parameter)", steps[['download']]$target_name)
+    }
+)
+  # New Step "tally_step"
+  tally_step <- create_task_step(
+    step_name = 'tally',
+    target_name = function(task_name, step_name, ...){
+      sprintf("%s_tally", task_name)
+      },
+    command = function(steps, ...){
+      sprintf("tally_site_obs(site_data = %s)", steps[['download']]$target_name)
```

Comment:
Oh, I meant to follow up on this in our earlier chat - the other way you _could_ do this would be 
```
sprintf("tally_site_obs(site_data=%s_data)", task_name)
```
which still uses `sprintf` but doesn't use `steps`.

That said, I prefer the way you've done it here.